projects/plugins/jetpack/_inc/blogging-prompts.php

```diff
@@ -18,3 +20,186 @@ function jetpack_blogging_prompts_add_meta_data( $keys ) {
 	$keys[] = '_jetpack_blogging_prompt_key';
 	return $keys;
 }
+
+add_filter( 'rest_api_allowed_public_metadata', 'jetpack_blogging_prompts_add_meta_data' );
+
+/**
+ * Sets up a new post as an answer to a blogging prompt.
+ *
+ * Called on `wp_insert_post` hook.
+ *
+ * @param int $post_id ID of post being inserted.
+ * @return void
+ */
+function jetpack_setup_blogging_prompt_response( $post_id ) {
+	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	$prompt_id = isset( $_GET['answer_prompt'] ) && absint( $_GET['answer_prompt'] ) ? absint( $_GET['answer_prompt'] ) : false;
+
+	if ( ! jetpack_is_new_post_screen() || ! $prompt_id ) {
+		return;
+	}
+
+	if ( jetpack_is_valid_blogging_prompt( $prompt_id ) ) {
+		update_post_meta( $post_id, '_jetpack_blogging_prompt_key', $prompt_id );
+		wp_add_post_tags( $post_id, 'dailyprompt' );
+	}
+}
+
+add_action( 'wp_insert_post', 'jetpack_setup_blogging_prompt_response' );
+
+/**
+ * Utility functions.
+ */
+
+/**
+ * Retrieve a daily blogging prompt from the wpcom API and cache it.
+ *
+ * @return stdClass[] Array of blogging prompt objects.
+ */
+function jetpack_get_daily_blogging_prompts() {
```

Comment:
I think we should allow a timestamp to be passed into this function. When rendering notifications, we will need to get the prompt to match the note timestamp similar to [here](https://opengrok.a8c.com/source/xref/wpcom/wp-content/mu-plugins/writing-prompts/notification.php?r=cea1b20d#79).

Reply:
I've modified so you can specify a unix timestamp--let me know if that will work!